### PR TITLE
Remove Google Drive projects with no saved forms

### LINKED ISSUE #5402: Disable or remove Google Drive projects
We want to remove Google Drive projects that have no unsent forms and "disable" ones that have unsent forms.

## Acceptance

- [x] Given I've created a Google Drive project in an earlier version of Collect
And it's set as my current project
And I've also created a ODK server project
When I upgrade to and launch v2023.3
Then my current project changes to the ODK server project
And the Google Drive project is gone (in the UI and on disk)

- [x] Given I've created a Google Drive project in an earlier version of Collect
And it's set as my current project
And I've not created any other projects
When I upgrade to and launch v2023.3
Then I see the first launch screen
And the Google Drive project is gone (in the UI and on disk)

- [x] Given I've created a Google Drive project in an earlier version of Collect
And I have unsent submissions (finalized or non finalized)
When I upgrade to and launch v2023.3
Then the Google Drive project is still there
And the starting new forms should be blocked (there should be a dialog displayed after trying to fill a new form)
And there is a banner with a warning about the project being deleted in the next release (text TBD)

- [x] Given I've created a Google Drive project in an earlier version of Collect
And I have unsent submissions (finalized or non finalized)
When I upgrade to and launch v2023.3
And I click learn more on the warning banner (taking me to the Google Drive deprecation announcement)
And I return to the app
Then the warning banner is still there

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #5402 

#### What has been done to verify that this works as intended?
I've tested the changes manually and improved automated tests.

#### Why is this the best possible solution? Were any other approaches considered?
There is nothing important to discuss here.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
Please test removing Google Drive projects automatically according to the acceptance criteria described in the issue.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/getodk/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/getodk/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/getodk/collect/blob/master/docs/CODE-GUIDELINES.md#ui-components-style-guidelines)

### CODE REVIEW

**Comment by grzesiek2010** [comment]
@seadowg @alyblenkin This was not clearly described in the issue but I updated the banner in a way that it is always visible so I removed the `Dismiss` button too. I think that's ok since it's the last version before removing the functionality completely but wanted to confirm with you.

**Comment by alyblenkin** [comment]
Hey @grzesiek2010,  I agree with removing the dismiss button. It would be annoying, but we want to make it clear that it's going away soon!

Small design tweaks:
- Could we make the background of the message the same light blue that we use for the end of the finalization flow, so it reads as a warning and draws their attention to the top?
- I also think the icon we use in the finalization flow would be helpful to have at the beginning of the sentence. 

These are not show-stoppers, so if it's too much effort, I think we can ship it! 

image

**Comment by grzesiek2010** [comment]
Yeah, we can do that it will be like:
Screenshot_1689811033

**Comment by alyblenkin** [comment]
Looks great!

**Review by seadowg — CHANGES_REQUESTED** [review]
This does look like it makes changes so that the "Start new form" button is hidden for forms that still have unsent forms:

> Given I've created a Google Drive project in an earlier version of Collect
And I have unsent submissions (finalized or non finalized)
When I upgrade to and launch v2023.3
Then the Google Drive project is still there
And the "Start new form" button is gone in that project
And there is a banner with a warning about the project being deleted in the next release (text TBD)

**Comment by grzesiek2010** [comment]
@seadowg you are right I've missed the step with that button but now as I think about it I might have a better solution...
hiding the button wouldn't be perfect because starting new forms would be still possible using content providers. Taking that into account I think we should block starting new forms in `FormUriActivity` instead like we do in other cases where a form should not be started. Please take a look at the last commit and let me know if you agree https://github.com/getodk/collect/pull/5675/commits/b03fe755b804c64fced1c7eca754dad8a410bfa6

**Comment by grzesiek2010** [comment]
@seadowg I think you might have missed my question

**Comment by seadowg** [comment]
> @seadowg I think you might have missed my question

No I got it. Just behind on replies. Will get to it soon hopefully!

File: collect_app/src/androidTest/java/org/odk/collect/android/feature/instancemanagement/SendFinalizedFormTest.kt
Comment (seadowg) [inline collect_app/src/androidTest/java/org/odk/collect/android/feature/instancemanagement/SendFinalizedFormTest.kt]: I think we should keep this test and use `copyInstance` (like in `InstancesAdbTest`) to set up the submission instead of deleting it.

**Review by seadowg — CHANGES_REQUESTED** [review]

**Comment by grzesiek2010** [comment]
@getodk/testers 
`v2023.1` was the last version where adding GD project was possible so I think you can use that version and then try to upgrade to this pr.
You can also try upgrading from `v2023.1` to `v2023.2` (where adding new GD projects was removed) and then this pr to make sure everything is fine but I would just perform one test like that and all the other cases verify upgrading directly `v2023.1` to this pr.

**Comment by dbemke** [comment]
Can you attach app builds that allow accessing Google Drive and upgrading versions? I tried upgrading a released build (2023.1) from github to CircleCi  build from the PR - upgrade was impossible. I tried building my own app in AS (selfsigned, debug) and then "unable to make API calls"  so I'm not able to download forms from GD.

**Comment by grzesiek2010** [comment]
It doesn't work because you would need to [Enable the Google Drive API](https://github.com/getodk/collect#using-apis-for-local-development) but ok I can produce apks this will be faster:
https://drive.google.com/file/d/1q8hEWWWgD0FaE7JuyQe4RNX5Gapy63r9/view?usp=sharing

**Comment by dbemke** [comment]
We installed the app (from the drive) but still API calls are unavailable.

File: collect_app/src/androidTest/java/org/odk/collect/android/feature/instancemanagement/SendFinalizedFormTest.kt
Comment (grzesiek2010) [reply collect_app/src/androidTest/java/org/odk/collect/android/feature/instancemanagement/SendFinalizedFormTest.kt]: Right, done.

**Review by seadowg — APPROVED** [review]

**Comment by dbemke** [comment]
I created a GD project in v2023.1.2 and upgraded to the PR build and when I tap "Download" (in a GD project) there is a crash.
crashDownload

**Comment by dbemke** [comment]
Should a finalized form (in GD project upgraded to the PR) be editable? In "Ready to send" it's possible to view the form but not edit it.

**Comment by dbemke** [comment]
There is a crash on sending (in "Ready to send", send selected) the finalized form in a GD project in the PR version.

**Comment by srujner** [comment]
Both Crashes also appear on Android 13.

**Comment by grzesiek2010** [comment]
I was able to reproduce those crashes and it turns out it's a problem with code shrinking caused not by this pr but by upgrading Gradle about a month ago. The problem should be fixed.

> Should a finalized form (in GD project upgraded to the PR) be editable? In "Ready to send" it's possible to view the form but not edit it.

Yeah but for now it will work in this way (without editing). The change that introduce the grace period was merged to v2023.2.x but not to master and that's why in this pr editing finalized forms is not possible.

**Comment by grzesiek2010** [comment]
@lognaturel could you prepare a new apk for testing with the newest changes?

**Comment by dbemke** [comment]
If the user tries to reconfigure a GD project with a QR code with GD project the server settings change to demo.getodk.org. In the PR version changing the type to GD is impossible so as a result the user is not able to send submissions any more. The issue is also present in 2023.1.2 but in that version it’s still possible to set the server as GD again.

**Steps to reproduce:**
1. Install 2023.1.2 and create a GD project.
2. Go to "Get Blank Form” and download a form.
3. Go to "Fill Blank form” and mark form as finalized, save and exit the form
4. (Update Collect - optional)
5. On other device create a GD project and create a QR code via reconfigure QR code.
6. On the first device scan a QR code with Google Drive project.
7. Check the Server settings.
8. Try to send the form

**Comment by grzesiek2010** [comment]
> If the user tries to reconfigure a GD project with a QR code with GD project the server settings change to demo.getodk.org. In the PR version changing the type to GD is impossible so as a result the user is not able to send submissions any more. The issue is also present in 2023.1.2 but in that version it’s still possible to set the server as GD again.

We didn't discuss such a case. Do you think we should block reconfiguring projects (if it is a GD project)? or maybe leave it as is and ignore this case? I'm somewhat torn but if I had to choose I would block reconfiguring, @seadowg @alyblenkin

**Comment by seadowg** [comment]
> We didn't discuss such a case. Do you think we should block reconfiguring projects (if it is a GD project)? or maybe leave it as is and ignore this case?

Wow yeah that's an annoying scenario to get into to. I think I'd lean towards blocking as well.

**Comment by seadowg** [comment]
Oh and just to add: if that's the only thing blocking this, I reckon we merge and create an issue rather than holding this PR any longer.

**Comment by dbemke** [comment]
In the PR version there is a crash of the app on sending a form if the fallback submission URL is wrong.

**Comment by dbemke** [comment]
We finished testing the PR. 
Scenarios from the acceptance criteria work as expected. 
Fixed issues will be tested after merging this PR and PR #5521

**Pull request merged**
